### PR TITLE
py3.7 - py3.9 support

### PR DESCRIPTION
**Pull request by ChristianBesp**
Reworked and updated code and tests to run on current (3.7, 3.8, 3.9) python versions.

TBD: build wheels and update on pypi

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    - master
+    - '*rc'
+  pull_request:
+    branches:
+    - master
+
+jobs:
+  tests:
+    name: Python ${{matrix.python-version}}
+    runs-on: ubuntu-20.04
```
Comment (DavidLP) [inline .github/workflows/tests.yml:15]: Nice, did not really use github actions thus far. Can we also test under windows or OSX with this?

**Review by DavidLP — CHANGES_REQUESTED** [review]
Very nice, indeed. I guess this package is in good hands :-)

**Comment by DavidLP** [comment]
Autodeploy to pypi on successfull tag build is a feature that was working once...

File: .github/workflows/tests.yml (hunk 1 above)
Comment (ChristianBesp) [reply .github/workflows/tests.yml:15]: Yes, both macOS and Windows Images are available but will count more (x10 and x2)towards the available build minutes. I will integrate those and see

File: README.md [hunk 2]
```diff
@@ -1,26 +1,26 @@
-# pylandau [![Build Status](https://travis-ci.org/SiLab-Bonn/pyLandau.svg?branch=master)](https://travis-ci.org/SiLab-Bonn/pyLandau) [![Build Status](https://ci.appveyor.com/api/projects/status/github/SiLab-Bonn/pyLandau)](https://ci.appveyor.com/project/DavidLP/pyLandau)
+# pylandau
+![example branch parameter](https://github.com/Silab-Bonn/pylandau/actions/workflows/tests.yml/badge.svg?branch=2.1.2rc)
```
[the remark this answers is not in the document]
Comment (ChristianBesp) [reply README.md:2]: Unfortunately, Github does not provide such a parameter. Could be done with some pre-commit hook, but for now, manually seems easier to me

File: .github/workflows/tests.yml (hunk 1 above)
Comment (ChristianBesp) [reply .github/workflows/tests.yml]: Added Windows and macOS CI runs

**Comment by ChristianBesp** [comment]
@DavidLP You want to check? It says there are still changes requested (although I should have resolved all). Otherwise I would merge

**Review by DavidLP — APPROVED** [review]

**Comment by DavidLP** [comment]
Awesome!

**Pull request merged**